### PR TITLE
Remove code that potentially let DV get into failed state with Restar…

### LINKED ISSUE #642: Pod import's pod status flaps between Running and Failed breaking retry logic
/kind bug

**What happened**:

#495 introduced special logic that marks a pod as failed on the PVC annotation when the pod has not in fact failed.  Instead we're interpreting the Pod's phase as failed because a container failed, even though the pod's status is still Running. 

We have 'restartPolicy: OnFailure' set on our import pods.  This means a container can fail, but will be retried. That's why the pod's phase remains Running. 

So, what we have now is a PVC that will move to a finalized "Failed" state, and then flap back over to Running or Succeeded later on if a subsequent retry of the pod's container completes successfully. 

**What you expected to happen**:

Once the PVC is marked as Failed, I wouldn't expect it to magically go back to running or succeeded later on. 

**Anything else we need to know?**:

This impacts DataVolumes as well. Since DataVolumes derive their phase from the PVC's annotation's pod phase, DataVolumes flap back and forth between failed and succeeded as well under some circumstances. This issue was discovered because it is causing issues with DataVolume integration with KubeVirt.

### PR DESCRIPTION
**Pull request by awels**
…tOnFailure policy.

Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:
In some cases we would mark an import failed while the pod cannot get into that state while restartPolicy is RestartOnFailure. The pod status will remain running even if the container is restarting. This caused the DV to potentially get into failed and back again which is not a valid state transition. This PR removes the code as it doesn't do what we want anyway.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #642 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/import-controller.go [hunk 1]
```diff
@@ -229,9 +229,8 @@ func (r *ImportReconciler) updatePvcFromPod(pvc *corev1.PersistentVolumeClaim, p
 	}
 
 	anno[AnnImportPod] = string(pod.Name)
-	if !scratchExitCode {
-		r.updatePvcPhaseFromPod(anno, pod)
-	}
+	// Even if scratch space is needed, the pod state will still remain running, until the new pod is started.
+	anno[AnnPodPhase] = string(pod.Status.Phase)
```
Comment (mhenriks) [inline pkg/controller/import-controller.go]: I think that it is good to have an annotation that accurately reflects the pod phase.  But now there is no way for the user to know if there was a problem.  Perhaps they entered the url wrong?  This hit me after looking at the changes made to the unit tests.  Imports will never "fail" now.  May have to think about this a little more.

Comment (awels) [reply pkg/controller/import-controller.go]: Remember we emit events when we have a failure, and that event is noted in the PVC. And you are correct, if we are fully eventually consistent, then we will never 'fail' the import, we will keep retrying forever.

Comment (mhenriks) [reply pkg/controller/import-controller.go]: Do the tests validate that the events are generated on failure?  If so then I ay be fine with this?

Comment (awels) [reply pkg/controller/import-controller.go]: https://github.com/kubevirt/containerized-data-importer/blob/master/tests/datavolume_test.go#L116-L125 yes they do.

Comment (awels) [reply pkg/controller/import-controller.go]: In fact it verifies both the event type and event message is what we are expecting when dealing with failures.

**Pull request merged**